File: 백준/Silver/10773. 제로/제로.cc
#include <iostream>
#include <stack>
using namespace std;

stack<int> s;
int K;
int num;
int stack_size;
int sum = 0;

int main() {

	cin >> K;

	for (int i = 0; i < K; i++) {
		cin >> num;
		if (num == 0) {
			s.pop();
		}
		else
		{
			s.push(num);
		}
	}

	stack_size = s.size();

	for (int i = 0; i < stack_size; i++) 
	{
		sum += s.top();
		s.pop();
	}

	cout << sum;
}

File: 백준/Silver/10773. 제로/제로.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        int K = int.Parse(Console.ReadLine());
        Stack<int> stack = new Stack<int>();

        for (int i = 0; i < K; i++)
        {
            int num = int.Parse(Console.ReadLine());

            if (num == 0)
            {
                stack.Pop();
            }
            else
            {
                stack.Push(num);
            }
        }

        long sum = 0;
        foreach (int n in stack)
        {
            sum += n;
        }

        Console.WriteLine(sum);
    }
}
